### PR TITLE
meta(engine): rename some meta name

### LINKED ISSUE #6938: meta(engine):  Rename some old meta name to a more proper one
### Is your feature request related to a problem?

No

### Describe the feature you'd like

Rename some old meta name to a more proper one , like `MasterMetaKVData`, `StatusCode`、`Code`

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #6938 

### What is changed and how it works?
1. Rename `MasterMetaKVData`  => `MasterMeta`
2. Rename `StatusCode` => `State`
3. Rename `Code` => `State`
4. Rename `WorkerStatusXXX` => `WorkerStateXXX`
5. Rename `MasterStatusXXX` => `MasterStateXXX`
6. Rename `Tp` => `Type`

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/framework/metadata/metadata.go [hunk 1]
```diff
@@ -81,34 +81,34 @@ func (c *MasterMetadataClient) Delete(ctx context.Context) error {
...
 		return nil, errors.Trace(err)
 	}
 	return meta, nil
 }
 
-// WorkerMetadataClient provides all ways to manage metadata of all workers
+// WorkerStatusdataClient provides all ways to manage metadata of all workers
 // belonging to a given master
-type WorkerMetadataClient struct {
+type WorkerStatusdataClient struct {
```
Comment (amyangfei) [inline engine/framework/metadata/metadata.go:94]: This name is strange.

Comment (maxshuang) [reply engine/framework/metadata/metadata.go]: Rename to `WorkerStatusClient`

**Review by amyangfei — APPROVED** [review]

File: engine/servermaster/jobmanager.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ type JobManager interface {
 	pb.JobManagerServer
 
 	GetJobMasterForwardAddress(ctx context.Context, jobID string) (string, error)
-	GetJobStatuses(ctx context.Context) (map[frameModel.MasterID]frameModel.MasterStatusCode, error)
+	GetJobStatuses(ctx context.Context) (map[frameModel.MasterID]frameModel.MasterState, error)
```
Comment (gozssky) [inline engine/servermaster/jobmanager.go:54]: Can we have any plan to rename related variable names or function names like `GetJobStatuses`?

File: engine/pkg/externalresource/manager/interfaces.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ type ExecutorInfoProvider interface {
 }
 
 // JobStatus describes the a Job's status.
-type JobStatus = frameModel.MasterStatusCode
+type JobStatus = frameModel.MasterState
```
Comment (gozssky) [inline engine/pkg/externalresource/manager/interfaces.go:39]: Ditto for `JobStatus`.

File: engine/executor/dm/worker.go [hunk 4]
```diff
@@ -245,12 +245,12 @@ func (w *dmWorker) tryUpdateStatus(ctx context.Context) error {
 func (w *dmWorker) workerStatus(ctx context.Context) frameModel.WorkerStatus {
 	var (
 		stage       = w.getStage()
-		code        frameModel.WorkerStatusCode
+		code        frameModel.WorkerState
```
Comment (gozssky) [inline engine/executor/dm/worker.go:248]: Ditto for `code`.

File: engine/servermaster/jobmanager.go (hunk 2 above)
Comment (maxshuang) [reply engine/servermaster/jobmanager.go:54]: To control the size of this pr, we leave it to another pr.

File: engine/pkg/externalresource/manager/interfaces.go (hunk 3 above)
Comment (maxshuang) [reply engine/pkg/externalresource/manager/interfaces.go:39]: Ditto.

File: engine/executor/dm/worker.go (hunk 4 above)
Comment (maxshuang) [reply engine/executor/dm/worker.go:248]: ditto.

**Review by gozssky — APPROVED** [review]

**Pull request merged**